File: 118-pascals-triangle/118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    var pascal = [];
    for (var i = 0; i < numRows; i++) {
        pascal[i] = [];
        pascal[i][0] = 1;
        for (var j = 1; j < i; j++) {
            pascal[i][j] = pascal[i-1][j-1] + pascal[i-1][j]
        }
        pascal[i][i] = 1;
    }
    return pascal;
};

File: 118-pascals-triangle/118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var result = new int[numRows][];
        
        for(int i = 0; i < numRows; i++) {
            result[i] = new int[i + 1];
            result[i][0] = 1;
            result[i][i] = 1;

            for(int j = 1; j < i; j++) {
                result[i][j] = result[i - 1][j - 1] + result[i - 1][j];
            }
        }
        
        return result;
    }
}